# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1744314264 -7200
# Parent  5f3ed0a3eec0cbe387dbcc96fd8122ba61c0f8a1
NNN22 - Follow-up to bug 1836511: Include folder name in new mail alert title.

diff --git a/mailnews/base/content/newmailalert.js b/mailnews/base/content/newmailalert.js
--- a/mailnews/base/content/newmailalert.js
+++ b/mailnews/base/content/newmailalert.js
@@ -27,10 +27,17 @@ function prefillAlertInfo() {
   var totalNumNewMessages = newMsgKeys.length;
   const message = document
     .getElementById("bundle_messenger")
     .getString("newMailAlert_message");
+  // Note that the system notification uses `folder.server.prettyName`
+  // which should return the same as `folder.server.rootFolder.prettyName`.
+  let folderInfo = folder.server.prettyName;
+  // For non-inbox folders add the folder name.
+  if (!(folder.flags & Ci.nsMsgFolderFlags.Inbox)) {
+    folderInfo += ` (${folder.prettyName})`;
+  }
   label.value = PluralForm.get(totalNumNewMessages, message)
-    .replace("#1", folder.server.rootFolder.prettyName)
+    .replace("#1", folderInfo)
     .replace("#2", totalNumNewMessages);
 
   // <folder-summary> handles rendering of new messages.
   var folderSummaryInfoEl = document.getElementById("folderSummaryInfo");
diff --git a/mailnews/base/src/MailNotificationManager.sys.mjs b/mailnews/base/src/MailNotificationManager.sys.mjs
--- a/mailnews/base/src/MailNotificationManager.sys.mjs
+++ b/mailnews/base/src/MailNotificationManager.sys.mjs
@@ -256,13 +256,18 @@ export class MailNotificationManager {
    * @param {number} numNewMessages - The count of new messages.
    * @returns {string} The alert title.
    */
   _getAlertTitle(folder, numNewMessages) {
+    let folderInfo = folder.server.prettyName;
+    // For non-inbox folders add the folder name.
+    if (!(folder.flags & Ci.nsMsgFolderFlags.Inbox)) {
+      folderInfo += ` (${folder.prettyName})`;
+    }
     return this._bundle.formatStringFromName(
       numNewMessages == 1
         ? "newMailNotification_message"
         : "newMailNotification_messages",
-      [folder.server.prettyName, numNewMessages.toString()]
+      [folderInfo, numNewMessages.toString()]
     );
   }
 
   /**
